AdobeStockAsset/Model/Config/Backend/AdobeStockApi.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\AdobeStockAsset\Model\Config\Backend;
+
+use Magento\AdobeStockClient\Model\Config ;
+use Magento\Framework\Exception\LocalizedException;
+use AdobeStock\Api\Models\SearchParameters;
+use AdobeStock\Api\Request\SearchFiles as SearchFilesRequest;
+use AdobeStock\Api\Client\AdobeStock;
+
+
+/**
+ * Class AdobeStockApi
+ * @package Magento\AdobeStockAsset\Model\Config\Backend
+ */
+class AdobeStockApi
+{
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * AdobeStockApi constructor.
+     * @param Config $adobeConfig
+     */
+    public function __construct(
+        Config $adobeConfig
+    )
+    {
+        $this->config = $adobeConfig;
+    }
+
+    /**
+     * @param \Magento\Config\Model\Config\Backend\Encrypted $subject
+     * @throws LocalizedException
+     */
+    public function beforeBeforeSave(\Magento\Config\Model\Config\Backend\Encrypted $subject)
+    {
+        $value = $subject->getValue();
+        $this->testConnection($value);
+    }
+    /**
+     * @inheritdoc
+     */
+    public function testConnection($value)
```

Comment:
We already have `testConnection` method here: `\Magento\AdobeStockClient\Model\Client::testConnection`

Reply:
Hello @sidolov 

We need to refactor this function if we want to use because of this function we can test after system configuration value has been saved but as per this task requirement, we need to check before saving the system configuration. You can check `\Magento\AdobeStockClient\Model\Client::getClient` on line 173. Please check let know how you want to implement before saving the system configuration

Thanks